### PR TITLE
Speclite zeropoints

### PR DESCRIPTION
**Pull request by mpaillassa**
Here is the PR related to speclite zeropoints #48.

I wrote a first script that can be run separately to compute zeropoints with speclite (and current effective area) for available surveys/filters and to compare them with current values (just by printing).

Does this seem ok at least in the general way to implement it or would you have something else in mind @aboucaud ?

### CODE REVIEW

File: galcheat/zeropoints.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        if survey_name in speclite_available_surveys:
+            survey = get_survey(survey_name)
+            print(survey_name, ":")
+
+            speclite_prefix = speclite_survey_prefixes[
+                speclite_available_surveys.index(survey_name)
+            ]
+
+            survey_filters = survey.get_filters()
+            for filt_name in survey_filters:
```
Comment (aboucaud) [inline galcheat/zeropoints.py:35]:
```suggestion
            for filt_name in survey.available_filters:
```

File: galcheat/zeropoints.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                speclite_filt_name = f"{speclite_prefix}{filt_name}"
+                zp_24 = (
+                    calculate_zero_point(speclite_filt_name)
+                    * survey.effective_area
+                    * 1
```
Comment (aboucaud) [inline galcheat/zeropoints.py:40]: Is this the 1 e- ? There is a unit for that (just in case).

**Review by aboucaud — COMMENTED** [review]
Really cool !

I made small comments for improving the code but this looks really nice.

I would suggest to put such script in a `scripts` folder at the base if the repo, because this is not meant to be a part of the `galcheat` module IMO. Also we will put a dedicated requirements file for the scripts with `speclite` in it to avoid adding a dependency to `galcheat`.

Unless you have a specific thing in mind.

File: galcheat/zeropoints.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import math
+
+from astropy import units as u
+from speclite.filters import ab_reference_flux, load_filter
+
+from galcheat.helpers import available_surveys, get_survey
+
+speclite_available_surveys = ["DES", "Euclid_VIS", "HSC", "Rubin"]
+speclite_survey_prefixes = ["decam2014-", "Euclid-", "hsc2017-", "lsst2016-"]
```
[the remark this answers is not in the document]
Comment (mpaillassa) [reply galcheat/zeropoints.py]: Yes !

File: galcheat/zeropoints.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+
+def calculate_zero_point(band_name, B0=24):
```
[the remark this answers is not in the document]
Comment (mpaillassa) [reply galcheat/zeropoints.py]: Hm in fact I took this function from [here](https://github.com/LSSTDESC/WeakLensingDeblending/issues/19). I am not totally sure but I think we could call this `B0` variable `reference_mag`.

File: galcheat/zeropoints.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    """Compute the zeropoint of a given filter with speclite"""
+    filt = load_filter(band_name)
+    return (filt.convolve_with_function(ab_reference_flux) * 10 ** (-0.4 * B0)).to(
```
[the remark this answers is not in the document]
Comment (mpaillassa) [reply galcheat/zeropoints.py]: Yes I decomposed it in several steps.

File: galcheat/zeropoints.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        1 / (u.s * u.m**2)
+    )
+
+
+def check_zeropoints():
+    """Recompute the zeropoints with speclite and the effective area
+    and compare them to their current values
+    """
```
[the remark this answers is not in the document]
Comment (mpaillassa) [reply galcheat/zeropoints.py]: Done !

File: galcheat/zeropoints.py (hunk 2 above)
Comment (mpaillassa) [reply galcheat/zeropoints.py]: So here I am not totally sure neither. The `calculate_zero_point` function outputs the zero point in 1/(s.m^2). Multiplying by the effective area gives the zero point in 1/s, which I interpret as a throughput per second. Then I multiply by 1s to obtain for the duration of 1s, convert it to our magnitude convention and set it to a magnitude unit. Does it make sense ?

File: scripts/zeropoints.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+        speclite_prefix = speclite_survey_prefixes[survey_name]
+
+        for filt_name in survey.available_filters:
+            speclite_filt_name = f"{speclite_prefix}{filt_name}"
+            zp_24 = (
+                calculate_zero_point(speclite_filt_name)
+                * survey.effective_area
+                * 1
+                * u.s
```
Comment (aboucaud) [inline scripts/zeropoints.py:42]:
```suggestion
                * 1 * u.s
```
makes more sense to me that way

File: galcheat/zeropoints.py (hunk 2 above)
Comment (aboucaud) [reply galcheat/zeropoints.py]: Committed a slight change to put the second next to the one to avoid confusion.

File: scripts/zeropoints.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+        print(survey_name, ":")
+
+        speclite_prefix = speclite_survey_prefixes[survey_name]
+
+        for filt_name in survey.available_filters:
+            speclite_filt_name = f"{speclite_prefix}{filt_name}"
+            zp_24 = (
+                calculate_zero_point(speclite_filt_name)
+                * survey.effective_area
+                * 1 * u.s
```
Comment (aboucaud) [inline scripts/zeropoints.py:41]:
```suggestion
                * (1 * u.s)
```

**Pull request merged**